lib/ansible/modules/cloud/amazon/code_commit.py

```diff
@@ -0,0 +1,144 @@
+'''
+
+RETURN = '''
+results:
+    description: A complex type that contains information about results
+    returned: changed
+    type: dict
+
+'''
+
+EXAMPLES = '''
+# Create new A repository
+- code_commit:
+    name: repo
+    state: present
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # Handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import (
+    boto3_conn,
+    boto_exception,
+    ec2_argument_spec
+)
+
+
+class CodeCommit(object):
+    def __init__(self, module=None):
+        self._module = module
+        self._connection = self._module.client('codecommit')
+        self._check_mode = self._module.check_mode
+
+    def process(self):
+        results = dict(changed=False)
+        changed = False
+        repository_exists = self._repository_exists()
+
+        if self._module.params['state'] == 'present' and not repository_exists:
+            changed = True
+            if not self._module.check_mode:
+                changed, results = self._create_repository()
+        if self._module.params['state'] == 'absent' and repository_exists:
+            changed = True
+            if not self._module.check_mode:
+                changed, results = self._delete_repository()
+        return changed, results
+
+    def _repository_exists(self):
+        try:
+            repository_exists = False
+            repositories = self._connection.list_repositories()['repositories']
+            for item in repositories:
+                if not repository_exists:
+                    repository_exists = self._module.params['name'] in item.values()
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self._module.fail_json_aws(e, msg="couldn't get repository")
+        return repository_exists
+
+    def _create_repository(self):
+        try:
+            results = self._connection.create_repository(
+                repositoryName=self._module.params['name'],
+                repositoryDescription=self._module.params['comment']
+            )
+            changed = True
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self._module.fail_json_aws(e, msg="couldn't create repository")
+        return changed, results
+
+    def _delete_repository(self):
+        try:
+            results = self._connection.delete_repository(
+                repositoryName=self._module.params['name']
+            )
+            changed = True
+        except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+            self._module.fail_json_aws(e, msg="couldn't delete repository")
+        return changed, results
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        name=dict(required=True),
+        state=dict(choices=['present', 'absent'], required=True),
+        comment=dict(default='')
+    ))
```

Comment:
AnsibleAWSModule automatically merges the argument_spec with ec2_argument_spec(), you don't have to do it manually.